c/zss.c

```diff
@@ -698,37 +698,68 @@ static void installWebPluginDefintionsService(WebPluginListElt *webPlugins, Http
   char errorBuffer[1024] = {0};
-  
-  Json *json = jsonParseFile(slh, serverConfigFile, errorBuffer, sizeof(errorBuffer));
-  if (json) {
-    if (jsonIsObject(json)) {
-      JsonObject *jsonObject = jsonAsObject(json);
-      JsonObject *logLevels = jsonObjectGetObject(jsonObject, "logLevels");
-      if (logLevels != NULL) {
-        JsonProperty *property = jsonObjectGetFirstProperty(logLevels);
-        while (property != NULL) {
-          if (!strcmp(jsonPropertyGetKey(property), pluginIdentifier)) {
-            int logLevel = jsonObjectGetNumber(logLevels, pluginIdentifier);
-            if (logLevel <= ZOWE_LOG_DEBUG3 || logLevel >= ZOWE_LOG_ALWAYS) {
-              return logLevel;
-            }
-            else {
-              break;
-            }
-          }
-          property = jsonObjectGetNextProperty(property);
-        }
+  int logLevel = ZOWE_LOG_SEVERE;
+
+  JsonObject *logLevels = jsonObjectGetObject(serverConfig, "logLevels");
+  if (logLevels != NULL) {
+    JsonProperty *property = jsonObjectGetFirstProperty(logLevels);
+    while (property != NULL) {
+      if (!strcmp(jsonPropertyGetKey(property), pluginIdentifier)) {
+        logLevel = jsonObjectGetNumber(logLevels, pluginIdentifier);
+        break;
       }
+      property = jsonObjectGetNextProperty(property);
     }
   }
-  
+
+  if (logLevel <= ZOWE_LOG_DEBUG3 || logLevel >= ZOWE_LOG_ALWAYS) {
+    return logLevel;
+  }  
   return ZOWE_LOG_INFO;
 }
 
+static char *formatDataServiceIdentifier(const char *unformattedIdentifier, size_t identifierLength, char idNameSeparator)
+{
+  char *dataServiceIdentifier = safeMalloc(identifierLength, "dataServiceIdentifier");
+  if (dataServiceIdentifier) {
+    memset(dataServiceIdentifier, 0, identifierLength);
+    strncpy(dataServiceIdentifier, unformattedIdentifier, identifierLength);
+    for (int i = 0; dataServiceIdentifier[i]; i++) {
+      if (dataServiceIdentifier[i] == '/') {
+        dataServiceIdentifier[i] = idNameSeparator;
+      }
+    }
+  }
+  return dataServiceIdentifier;
+}
+
+static void checkAndSetDataServiceLoglevel(JsonObject * serverConfig,
+                                           char *dataServiceIdentifier, 
+                                           uint64 loggingIdentifier) {
+  char errorBuffer[1024] = {0};
+  int logLevel = ZOWE_LOG_SEVERE;
+
+  JsonObject *logLevels = jsonObjectGetObject(serverConfig, "logLevels");
+  if (logLevels != NULL) {
+    JsonProperty *property = jsonObjectGetFirstProperty(logLevels);
+    while (property != NULL) {
+      if (!strcmp(jsonPropertyGetKey(property), dataServiceIdentifier)) {
+        logLevel = jsonObjectGetNumber(logLevels, dataServiceIdentifier);
+        break;
+      }
+      property = jsonObjectGetNextProperty(property);
+    }
+  }
+
+  if (logLevel <= ZOWE_LOG_DEBUG3 || logLevel >= ZOWE_LOG_ALWAYS) {
+    logSetLevel(NULL, loggingIdentifier, logLevel);
```

Comment:
Please call `logSetLevel` only when the corresponding key is found found in `server.json`.

Reply:
Resolved